tests/php/FilterExpressionParsing/FilterParserTest.php

```diff
@@ -69,6 +69,10 @@ public function provideParseCases() {
 				'{prop1:var1, prop2:var2}',
 				new ParseResult( [ '{prop1:var1, prop2:var2}' ], [] )
 			],
+			'method call with single argument' => [
+				'filter(var1)',
+				new ParseResult( [ 'var1' ], [ new FilterCall( 'filter', [] ) ] )
+			],
 			'variable with filter having no arguments' => [
 				'var1|filter',
 				new ParseResult( [ 'var1' ], [ new FilterCall( 'filter', [] ) ] )
```

Comment:
It’s weird how, just below this test, you can see that we actually have support for parenthesized calls already (`var1|filter(var2)`). (I had no idea Vue supported this, and we certainly don’t seem to use it in WikibaseLexeme – so much for only covering a subset of the Vue syntax that’s used by WikibaseLexeme…) But this seems to rely on the parser’s `$this->nowParsingFilterList` state: all parentheses before the first filter pipe are “grouping” parentheses, and all parentheses after the first filter pipe are “call” parentheses. I don’t think we can easily repurpose this code for parsing method calls, at least not while keeping support for filter syntax as well, so I think the regex hack is good enough for now.

Once we drop filter support entirely, I think we can remove the whole `FilterExpressionParsing` step, and instead move the regex into the `BasicJsExpressionParser`? (Which is quite basic indeed – lots of the syntax that the filter expression parser supports and tests, like arrays and objects, don’t actually work in the JS part as far as I can tell.)

Reply:
Yes, I noticed the existing syntax as well. But making use of that without breaking anything seemed like the more intricate challenge